### ISSUE TITLE
Add name argument to Model class to allow naming the output when using sympy expression

### ISSUE THREAD

**Issue by heikowestermann** [sandialabs/suncal#12]
Hey,

It would be great to have a `name` attribute for `Model` so that we can name the output ourselves and don't need to use the automatic naming. In case of a `str` expression the internal `sympify` will parse the name, but when using a sympy-type instead, the name cannot be set.
Currently the output will be named automatically f_1, f_2, etc.

I would be willing to provide a pull request if you allow external contributors.

**Comment by cdelker** [comment]
Good idea. I'd be happy to take a pull request if you're able to put it together.
